code/modules/power/lighting.dm

```diff
@@ -261,23 +256,26 @@
 		set_light(0)
 
 	active_power_usage = (luminosity * 10)
-	if(on != on_gs)
-		on_gs = on
+	if(light_on != on_gs)
+		on_gs = light_on
 	update_icon()
 
-
 // attempt to set the light's on/off status
 // will not switch on if broken/burned/empty
-/obj/machinery/light/proc/seton(s)
-	on = (s && status == LIGHT_OK)
+/obj/machinery/light/turn_light(mob/user, toggle_on)
+	if ((status != LIGHT_DISABLED) & (status != LIGHT_OK)) //Can't turn a broken light
+		return
```

Comment:
Shouldnt this have a return value?

Reply:
it won't get used anywhere

Reply:
Add one that makes sense anyway because you guarantee a return value in parentcalls